mesonbuild/interpreter/interpreter.py

```diff
@@ -145,7 +145,7 @@ def stringifyUserArguments(args: T.List[T.Any], quote: bool = False) -> str:
         return str(args)
     elif isinstance(args, str):
         return f"'{args}'" if quote else args
-    raise InvalidArguments('Function accepts only strings, integers, bools, lists, dictionaries and lists thereof.')
+    return args.display_repr()
```

Comment:
If you're changing something from an error to a valid function call, then it needs to emit a `FeatureNew` describing the newly allowed usage. This applies regardless of anything else.